src/odemis/gui/util/img.py

```diff
@@ -1205,9 +1207,19 @@ def spectrum_to_export_data(proj, raw, vp=None):
         ctx = cairo.Context(surface)
 
         if vp is not None:
+
             # Limit to the displayed ranges
             range_x = vp.hrange.value
             range_y = vp.vrange.value
+            # extend the range_y by 5% from both sides (range extended by 10% in total)
+            ext = (range_y[1] - range_y[0]) * 0.05
+            # Check if the range includes 0, after the extension, and compare 0 with the initial minimum and maximum
+            # values to replace them if needed.
+            if range_y[0] - ext <= 0 <= range_y[1] + ext:
+                if 0 < min(range_y):
+                    range_y = (0, max(range_y))
+                elif 0 > max(range_y):
+                    range_y = (min(range_y), 0)
```

Comment:
That's almost good. A corner-case that this doesn't handle is the "middle-case": if 0 is actually inside the range already (eg, values from -100 to 500). In such case, it would actually reduce the range. I think you could rewrite it this way:
```python
if range_y[0] - ext <= 0 <= range_y[1] + ext:
    range_y_and_0 = sorted([range_y[0], 0, range_y[1]])
    range_y = range_y_and_0[0], range_y_and_0[-1]

```